packages/react/src/components/SuggestionMenu/SuggestionMenuController.tsx

```diff
@@ -14,33 +14,70 @@ import { useUIPluginState } from "../../hooks/useUIPluginState";
 import { SuggestionMenu } from "./SuggestionMenu";
 import { SuggestionMenuWrapper } from "./SuggestionMenuWrapper";
 import { getDefaultReactSlashMenuItems } from "./getDefaultReactSlashMenuItems";
-import { DefaultReactSuggestionItem, SuggestionMenuProps } from "./types";
+import { GridSuggestionMenu } from "./GridSuggestionMenu";
+import { getDefaultReactEmojiPickerItems } from "./getDefaultReactEmojiPickerItems";
+import {
+  DefaultReactGridSuggestionItem,
+  DefaultReactSuggestionItem,
+  SuggestionMenuProps,
+} from "./types";
 
 type ArrayElement<A> = A extends readonly (infer T)[] ? T : never;
 
 type ItemType<GetItemsType extends (query: string) => Promise<any[]>> =
   ArrayElement<Awaited<ReturnType<GetItemsType>>>;
 
+// There's a lot going on with the typing here but tl;dr:
```

Comment:
To review (Y)

Reply:
I'm concerned the complexity of the code is not worth the wins we get out of it. The typing was already pretty complex (maybe too much), but now by adding a different "type" it increases too much I think. I think we've got three options:

1. Remove some of the typing tricks added in this PR for columns / grids etc. 
2. Revise the original code. Maybe we don't need all the defaults but it's better to explicitly pass in things like `defaultsuggestionmenuitems` from `BlockNoteDefaultUI`, etc.
3. Creating a separate GridSuggestionMenuController, that would be pretty much the same as the existing `SuggestionMenuController` but defaulting to emojis and grids

I think option 3 is the least effort while at least not making the code less readable from what it currently is. wdyt?

Reply:
Yeah agreed tbh, there will be a bit of code duplication but I think the other 2 options have bigger drawbacks (either worse typing or worse experience creating custom suggestion menus)